compiler/src/iree/compiler/Preprocessing/Common/test/preprocessing_pdl_patterns.mlir

```diff
@@ -0,0 +1,98 @@
+// RUN: iree-compile --iree-preprocessing-pdl-spec-filename=%p/linalg.pdl.mlir %s --compile-to=preprocessing | FileCheck %s
```

Comment:
This is fairly involved for a lit test. Could this be kept out of tree? Or maybe moved to a sample?

Reply:
Sure, we already have this out of tree, any thoughts on what I should show in the in-tree test then?

Reply:
For a lit test like this, any kind of proof of life is good. Like casting an i32 to index or something arbitrary, basically just to show that the patterns are being applied.

Reply:
I think I will move the test to samples, I am trying to show that the pass can be called with iree-compile so the transform has to be something that will lower e2e correctly. We already have tests for the pass itself.

Reply:
I added a small test here and the other test in mlp sampled.